From: Pedro Falcato <pedro.falcato@gmail.com>
Subject: selftests/mm: fix mseal's length
Date: Thu, 22 Aug 2024 00:20:19 +0100

We accidentally msealed too much, which could overrun and try to mseal
other regions.  This seems to be harmless (at least on top-down
architectures) due to various reasons all aligning, but may very well
cause spurious test failures to e.g bottom-up mmap architectures.

Link: https://lkml.kernel.org/r/vc4czyuemmu3kylqb4ctaga6y5yvondlyabimx6jvljlw2fkea@djawlllf45xa
Signed-off-by: Pedro Falcato <pedro.falcato@gmail.com>
Cc: Jeff Xu <jeffxu@chromium.org>
Cc: Kees Cook <kees@kernel.org>
Cc: Liam R. Howlett <Liam.Howlett@oracle.com>
Cc: Linus Torvalds <torvalds@linux-foundation.org>
Cc: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Cc: Michael Ellerman <mpe@ellerman.id.au>
Cc: Shuah Khan <shuah@kernel.org>
Cc: Vlastimil Babka <vbabka@suse.cz>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/mm/mseal_test.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/tools/testing/selftests/mm/mseal_test.c~selftests-mm-add-more-mseal-traversal-tests-fix-2
+++ a/tools/testing/selftests/mm/mseal_test.c
@@ -778,7 +778,7 @@ static void test_seal_mprotect_partial_m
 	FAIL_TEST_IF_FALSE(ptr != (void *)-1);
 
 	if (seal) {
-		ret = sys_mseal(ptr + page_size, size);
+		ret = sys_mseal(ptr + page_size, page_size);
 		FAIL_TEST_IF_FALSE(!ret);
 	}
 
@@ -1031,7 +1031,7 @@ static void test_seal_munmap_partial_acr
 	FAIL_TEST_IF_FALSE(ptr != (void *)-1);
 
 	if (seal) {
-		ret = sys_mseal(ptr + page_size, size);
+		ret = sys_mseal(ptr + page_size, page_size);
 		FAIL_TEST_IF_FALSE(!ret);
 	}
 
_
